libjwt/jwt.c

```diff
@@ -1094,6 +1094,24 @@ static int jwt_dump(jwt_t *jwt, char **buf, int pretty)
 	return ret;
 }
 
+char* jwt_dump_grants_str(jwt_t* jwt, int pretty) 
+{
+	char* out = NULL;
+	int err;
+
+	err = jwt_write_body(jwt, &out, pretty);
+	
+	if (err) {
+		errno = err;
+		if (out)
+			jwt_freemem(out);
+		out = NULL;
+	} else {
+		errno = 0;
```

Comment:
This should be set at the start of the function, before any other calls outside of this function. The reason being is to catch any errors set by outside libraries (e.g. I/O calls, etc.).

Reply:
> This should be set at the start of the function, before any other calls outside of this function. The reason being is to catch any errors set by outside libraries (e.g. I/O calls, etc.).

The logic of setting errno is the same as in jwt_dump_str.  Which value should be set at the start? EINVAL?